### DISCUSSION TITLE
How to insert a custom highlight color to a custom copy format?

### DISCUSSION THREAD

**Discussion by N3C2L** [RyotaUshio/obsidian-pdf-plus#105] [Q&A] [answered]
I want to add several customy copy formats with different pdf highlight colors.

I know there is this option here:

<img width="508" alt="pdf plus plus" src="https://github.com/RyotaUshio/obsidian-pdf-plus/assets/74585843/931d4699-eef7-4f93-9704-15c51ada48fc">

But is this option also intended for highlight colors and how is it used? Can you give me an example?

Because I saw from the callout example that only the callout in the annotation file get's colored with this color option and not the highlight color in the pdf?

**Comment by RyotaUshio** [comment, marked as answer]
Currently, highlight colors and copy formats are completely independent from each other, meaning that you can combine one format with any highlight color and you cannot bind a single format to each highlight color.

About the `color`/`colorName` template variable: it is the name of the color that is currently selected in the color palette. For example, if "Red" is selected then `color` becomes a string `"red"`.

An example is included in the preset copy format "Callout":
```
> [!{{calloutType}}|{{color}}] {{linkWithDisplay}}
> {{text}}
```
which is evaluated to, e.g
```
> [!PDF|red] [[file name.pdf#page=1&selection=0,3,2,1&color=red|file name, page 1]]
> This is the text selected in the PDF viewer
```
assuming red is selected in the color palette at the moment the link is copied.

**Comment by N3C2L** [comment]
Thanks, I got it.

Hmm, that makes think of something different. If these are independent from each other, then a macro function could be useful here? For example, one user can create a macro that will execute 
1. a custom copy format
2. a custom highlight color
3. auto-paste to annotation.

And these macros could then be added as commands.

**Reply by RyotaUshio** [reply]
That can be a nice addition, but to be honest, I'm not very sure in what situations we want to use different copy formats depending on different highlight colors. Can you elaborate a bit more about your workflow and specific copy formats?

PDF++'s templating system is so powerful that it can execute arbitrary JavaScript in `{{...}}` so your goal might be already achievable without adding new features.
